CMakeLists.txt

```diff
@@ -159,9 +169,21 @@ if(${KALDI_USE_PATCH_NUMBER})
     set(KALDI_VERSION "${KALDI_VERSION}.${KALDI_PATCH_NUMBER}")
 endif()
 
-get_third_party(openfst)
-set(OPENFST_ROOT_DIR ${CMAKE_BINARY_DIR}/openfst)
-include(third_party/openfst_lib_target)
+# get_third_party(openfst)
+# set(OPENFST_ROOT_DIR ${CMAKE_BINARY_DIR}/openfst)
+# include(third_party/openfst_lib_target)
+find_library(OpenFST_LIBRARY
+            NAMES fst 
+            PATHS ${CMAKE_SOURCE_DIR}/tools/openfst/lib 
+            REQUIRED)
+find_path(OpenFST_INCLUDE_DIR
+            NAMES "fst/fst.h" 
+            PATHS "${CMAKE_SOURCE_DIR}/tools/openfst/include"
+            REQUIRED)
+get_filename_component(OpenFST_LIBRARY_DIR ${OpenFST_LIBRARY} DIRECTORY CACHE)
+mark_as_advanced(FORCE OpenFST_INCLUDE_DIR OpenFST_LIBRARY OpenFST_LIBRARY_DIR)
+include_directories(${OpenFST_INCLUDE_DIR})
+link_directories(${OpenFST_LIBRARY_DIR})

```

Comment:
For this part, except suggestions, there are two changes:

1. I use `OpenFST_INCLUDE_DIR ` instead of `OpenFST_INCLUDE_DIRS`  to ensure naming consistency.
2. For `find_path`, there shall be NAMES field with `fst/fst.h`. `fst/fst.h` is because in Kaldi header file, FST header is included by `<fst/xxx.h>`. Using `fst.h` might results in compilation error.